website/src/docs/hotchocolate/performance/persisted-queries.md

```diff
@@ -2,12 +2,200 @@
+- HotChocolate.PersistedQueries.InMemory
+
+# Setup
+
+To Configure GraphQL server to use the persisted query pipeline.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        .AddRouting()
+        .AddGraphQLServer()
+        .AddQueryType<Query>()
+        .UsePersistedQueryPipeline();
+}
+```
+
+## Configure persisted queries on server (FileStorage)
+
+If we want to configure our GraphQL server to be able to handle persisted query requests. For this, we need to register the corresponding query storage and configure the persisted query request pipeline.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        .AddRouting()
+        .AddGraphQLServer()
+        .AddQueryType<Query>()
+        .AddReadOnlyFileSystemQueryStorage("./persisted_queries");
+}
+```
+
+By this HC will search for correspondent query Hash in folder `./persisted_queries`. It is important to store all exported clients persisted  queries in this folder using a specific name and format: `{Hash}.graphql`
+
+Example: `0c95d31ca29272475bf837f944f4e513.graphql`
+
+Now your server knows where to search for the requested query `Id` and loads it in memory in case of a successful match. 
+
+> ⚠️ **Note:** Don't forget to ensure if the storage directory exists and the server has particular access rights to it.
+
+## Configure persisted  queries on the client
+
+This step shows what you need to configure on the client-side or adjust on the server regarding clients. Example show Relay implementation. You can use any other GraphQL client like Apollo. The configuration is always client-related so follow offical client Docs. 
+
+[Docs Relay](https://relay.dev/docs/guides/persisted-queries/)
+
+[Docs Apollo](https://www.apollographql.com/docs/apollo-server/performance/apq/)
+
+### What can be different between clients
+
+- Hashing algorithm - Hot Chocolate server is configured to use by default the MD5 hashing algorithm. HotChocolate server comes out of the box with support for MD5, SHA1, and SHA256
+
+- Default Headers and Setup - HC expect the persisted `Id` to be named as` id` in the request header. Some docs like Relay show you to send it under `doc_id` this would not work with HC. You must adjust the client fetch function to use the correct variable. 
+
+### Hashing algorithm setup
+
+If adjustmed of algorithm is required you can do it using following sintax:
+
+ > **ℹ️**  For relay it is not needed. Relay use default MD5 Hash.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services
+        // Global Services
+        .AddRouting()
+        .AddMemoryCache()
+        .AddSha256DocumentHashProvider(HashFormat.Hex)
+        
+        // GraphQL server configuration
+        .AddGraphQLServer()
+        .AddQueryType<Query>()
+        .UsePersistedQueryPipeline()
+        .AddReadOnlyFileSystemQueryStorage("./persisted_queries");
+}
+```
+
```

Comment:
Let's add another section here to show how to add the security aspect and limit the server to only allow persisted queries. I will post you some code examples.

Reply:
Are you able to throw in an example @michaelstaib ? Thre's a TODO spot just lower. Then I think we can merge this. It'll be helpful for us at AutoGuru too, as this is next on our list for persisted queries now that we've got them all setup for the perf side of things.